src/pal/src/debug/debug.cpp

```diff
@@ -109,6 +109,7 @@ caches are coherent in hardware. For non-X86 architectures, this call
 usually maps to a kernel API to flush the D-caches on all processors.
 
 --*/
```

Comment:
Do all of the PAL APIs need to be exported - who is calling them?

Reply:
If I revert this file to `HEAD~1`, macOS shows the following warnings:

```c#
ld: warning: cannot export hidden symbol _FlushInstructionCache from ../../pal/src/libcoreclrpal.a(debug.cpp.o)
ld: warning: cannot export hidden symbol _OutputDebugStringA from ../../pal/src/libcoreclrpal.a(debug.cpp.o)
ld: warning: cannot export hidden symbol _OutputDebugStringW from ../../pal/src/libcoreclrpal.a(debug.cpp.o)
ld: warning: cannot export hidden symbol _DebugBreak from ../../pal/src/libcoreclrpal.a(debug.cpp.o)
ld: warning: cannot export hidden symbol _PAL_ProbeMemory from ../../pal/src/libcoreclrpal.a(debug.cpp.o)
```

I tried to keep the warning count exactly as master branch and did what xcode was telling me. 😊
https://api.travis-ci.org/v3/job/479888427/log.txt

Reply:
Ok, I see. All the APIs mentioned in these two files need to be exportable:

https://github.com/dotnet/coreclr/blob/master/src/dlls/mscoree/mscorwks_unixexports.src
https://github.com/dotnet/coreclr/blob/master/src/dlls/mscordac/mscordac_unixexports.src

It is just a subset of PAL APIs, but it would look pretty ad-hoc to have some APIs marked as DLLEXPORT and some not. I think it is ok to just mark all PAL APIs as DLLEXPORT for now. We can cleanup it up later. It may be nice to use PALEXPORT macro to mark the exportable PAL APIs - to make it clear that they are somewhat different from the regular exports.

Reply:
Marked visibility attribute to all PALAPIs in bf8ec08 with `DLLEXPORT`.

I tried defining and using `PALEXPORT` macro (2243c574cea9ea7290c621f3075d686956ac0c21), but couldn't make it work on Windows: there are places under src/pal (cruntime, safecrt) that are not marked with `PALAPI` and require export, then there are places outside pal which implicitly assume that some pal header would be exported before a non-pal header defining certain function. When it worked on macOS and Linux with no warnings, Windows was failing due to some non-pal files compiled in multiple projects with different configurations/macros, once with pal header and in second pass without it. Therefore, I have kept the `DLLEXPORT` for now. :)